### ISSUE TITLE
Database Deep Dive

### ISSUE THREAD

**Issue by OSP123** [OSP123/post-grad-curriculum#27]
- Going into database architecture; scaling (horizontal and vertical); other ORMs; using migrations and seeding; backing up databases; choosing the right service for database management and managing database.
- Deep dive into SQL
- Deep dive into noSQL

**Comment by DaC00P** [comment]
I can contribute Postgres and ActiveRecord/Rails material if we're interested.

**Comment by OSP123** [comment]
Ooh, that would be great if we can have a section for Rails and how to use migrations and seeding with that. Go for it!

**Comment by DaC00P** [comment]
Hmm, for Rails itself it may behoove us to make a separate story from the DB Deep Dive.
I will contribute Postgres stuff here though - I have a great example of how to do DB side full text searching with lexemes and how to utilize iLikes, etc.

**Comment by OSP123** [comment]
Yeah, I'm wondering how to logically separate the different tasks. Right now, each of these issues is a broad topic for discussion. 

Great! Yeah, I just want to make sure that we research and understand each aspect of what we're doing and have something to back it up, especially examples.

**Comment by BitwiseGuy** [comment]
I think perhaps a logical approach would be to have general database topics, which tend to be DB agnostic, but then link to sections that cover each topic by DB.

Start with basics like what is a DB, why you would use a DB versus in-memory collections, pluses and minuses of the different types of DBs like relational vs document, things like that. Then you can get into DB specifics like how to do migrations in the various DBs, popular ORMs, etc.

I think examples should start simple. One thing at a time. Too many tutorials try to cover 15 different topics to show a "simple" example, which just confuses most people.
